src/eureka/S5_lightcurve_fitting/differentiable_models/StarryModel.py

```diff
@@ -132,9 +132,43 @@ def setup(self):
             p = temp.per*(24.*3600.)
             Ms = ((2.*np.pi*a**(3./2.))/p)**2/const.G.value/const.M_sun.value
 
-            # Initialize star object
-            star = starry.Primary(starry.Map(udeg=self.udeg),
-                                  m=Ms, r=temp.Rs)
+            # check for spots and set parameters
+            if hasattr(self.parameters, 'spotrad0'):
+                spotrad = np.array([])
+                spotlat = np.array([])
+                spotlon = np.array([])
+                spotcon = np.array([])
+                for key in self.parameters.dict.keys():
+                    if key.startswith('spot'):
+                        if 'rad' in key:
+                            spotrad = np.append(spotrad, 
+                                                self.parameters.dict[key][0])
+                        elif 'lat' in key:
+                            spotlat = np.append(spotlat, 
+                                                self.parameters.dict[key][0])
+                        elif 'lon' in key:
+                            spotlon = np.append(spotlon, 
+                                                self.parameters.dict[key][0])
+                        elif 'con' in key:
+                            spotcon = np.append(spotcon, 
+                                                self.parameters.dict[key][0])
+                        elif 'rot' in key:
+                            starrot = self.parameters.dict[key][0]
+                        else:
+                            spotres = self.parameters.dict[key][0]
```

Comment:
This code is going to break for `mult_white` or shared fits. I recommend you make use of the `temp` object I defined above which should only include the parameters for the current channel